### PR TITLE
Add documentation for using git inside docker container

### PR DESCRIPTION
**Pull request by arjxn-py**
# Description

Add relevant instructions for developers in order to contribute to PyBaMM (setting up git configuration, setting a git remote

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [ ] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [ ] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ python run-tests.py --quick` (or `$ nox -s quick`).

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

**Comment by arjxn-py** [comment]
Related to #3312

**Review by Saransh-cpp — APPROVED** [review]
Looks good, thanks @arjxn-py!

**Review by agriyakhetarpal — APPROVED** [review]
Thanks, @arjxn-py! I guess we can leave out the part about configuring SSH and GPG keys with `git` because those who use them should be able to figure that out themselves

**Comment by Saransh-cpp** [comment]
@agriyakhetarpal any ideas why the readthedocs check is not running on this PR?

**Comment by Saransh-cpp** [comment]
Given that the readthedocs workflow is a required check, we cannot merge this PR (GitHub won't allow us). Maybe @tinosulzer can bypass the rule and merge this in.

**Comment by agriyakhetarpal** [comment]
Makes sense. If required we could trigger another build if someone pushes a further commit or rewrites the commit history

File: docs/source/user_guide/installation/install-from-docker.rst [hunk 1]
```diff
@@ -189,6 +189,28 @@ If you want to exit the Docker container's shell, you can simply type:
 
       exit
 
+
+Using Git Inside a Running Docker Container
+-------------------------------------------
+
+1. Setting up git configuration
```
Comment (Saransh-cpp) [inline docs/source/user_guide/installation/install-from-docker.rst:196]: We should add some information here. Something like - "You might require re-configuring git while running the docker container for the first time."?

Comment (arjxn-py) [reply docs/source/user_guide/installation/install-from-docker.rst:196]: Sure, looking into it.

**Review by Saransh-cpp — APPROVED** [review]
Looks good, thanks @arjxn-py!

**Pull request merged**